activestorage/app/controllers/concerns/active_storage/set_headers.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module ActiveStorage::SetHeaders #:nodoc:
+  extend ActiveSupport::Concern
+
+  private
+    def set_headers(blob)
+      expires_in 1.year, public: true
+
+      response.headers["Content-Type"] = blob.content_type
+      response.headers["Content-Disposition"] = ActionDispatch::Http::ContentDisposition.format(
+        disposition: params[:disposition] || "inline",
+        filename: blob.filename.sanitized
+      )
+    end
```

Comment:
`set_headers` is vague for what this method does. I think I’d pull `expires_in` out and rename this to `set_content_headers_from_blob` or similar:

```ruby
# Here:
def set_content_headers_from(blob)
  response.headers["Content-Type"] = blob.content_type
  response.headers["Content-Disposition"] = ActionDispatch::Http::ContentDisposition.format \
    disposition: params[:disposition] || "inline", filename: blob.filename.sanitized
end
```

```ruby
# activestorage/app/controllers/active_storage/blobs/proxy_controller.rb
def show
  expires_in 1.hour, public: true
  set_content_headers_from @blob
  stream @blob
end

private
  # Good candidate for hoisting to the base class and reusing elsewhere.
  def stream(blob)
    blob.download do |chunk|
      response.stream.write chunk
    end
  ensure
    response.stream.close
  end
```